extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMusicSearchExtractor.java

```diff
@@ -444,9 +460,15 @@ public String getName() throws ParsingException {
 
                         @Override
                         public String getUrl() throws ParsingException {
-                            final String url = getUrlFromNavigationEndpoint(info.getObject("doubleTapCommand"));
-                            if (!isNullOrEmpty(url)) {
-                                return url;
+                            final String playlistId = info.getObject("overlay")
+                                    .getObject("musicItemThumbnailOverlayRenderer")
+                                    .getObject("content")
+                                    .getObject("musicPlayButtonRenderer")
+                                    .getObject("playNavigationEndpoint")
+                                    .getObject("watchPlaylistEndpoint")
+                                    .getString("playlistId");
```

Comment:
@TobiGr Here is the code :)
```suggestion
                             final String playlistId = info.getObject("menu")
                                    .getObject("menuRenderer")
                                    .getArray("items")
                                    .getObject(4)
                                    .getObject("toggleMenuServiceItemRenderer")
                                    .getObject("toggledServiceEndpoint")
                                    .getObject("likeEndpoint")
                                    .getObject("target")
                                    .getString("playlistId");
```

Reply:
Thanks. I already extract the `playlistId` successfuly with the code you replaced. Did you get any errors with the current implementation?

Reply:
No but it's the audio version of albums ;)

Reply:
I can add that code as fallback extraction if the other method fails to extract the `playlistId`.

```suggestion
                            String playlistId = info.getObject("overlay")
                                    .getObject("musicItemThumbnailOverlayRenderer")
                                    .getObject("content")
                                    .getObject("musicPlayButtonRenderer")
                                    .getObject("playNavigationEndpoint")
                                    .getObject("watchPlaylistEndpoint")
                                    .getString("playlistId");
                            if (isNullOrEmpty(playlistId)) {
                                playlistId = info.getObject("menu")
                                        .getObject("menuRenderer")
                                        .getArray("items")
                                        .getObject(4)
                                        .getObject("toggleMenuServiceItemRenderer")
                                        .getObject("toggledServiceEndpoint")
                                        .getObject("likeEndpoint")
                                        .getObject("target")
                                        .getString("playlistId");
                            }
```

Reply:
With your code we'll get video version of albums but with my code the audio versions. So you should do the contrary if you want a fallback.

Reply:
I'm not exactly sure, but are you preferring the code that would fetch the audio versions with the fallback being the video versions? That seems the way to go.

Reply:
@opusforlife2 That's I want to say.

Reply:
Looks like we posted at the same time. 😆

Reply:
Uhm no. that's the same `playlistId`. the video version thing was because i forgot to use `music.youtube.com` instead of `youtube.com` when returning the complete URL. I fixed that earlier, but forgot to update the debug APK. Try this one. I cannot reproduce that behaviour anymore. 
[app-debug.zip](https://github.com/TeamNewPipe/NewPipeExtractor/files/5731212/app-debug.zip)